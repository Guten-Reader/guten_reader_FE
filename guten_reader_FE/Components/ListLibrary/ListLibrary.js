import React from 'react';
import { View, Text, Button, StyleSheet } from 'react-native';

export default function Library(props) {
  console.log('props', props)

  function displayBooks() {
    return props.books.map((book) => {
      return (
        <View style={styles.list}>
          <Text style={styles.listItem}>{book}</Text>
          <Button style={styles.button} onPress={props.downloadBook} title="READ" onPress={props.downloadBook}></Button>
        </View>
      )
    })
  }

    return (
      <View>
        {displayBooks()}
      </View>
    )
}

const styles = StyleSheet.create({
  list: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 20
  },
  listItem: {
   fontSize: 20
   },
   button: {
    fontSize: 20,
    color: '#53E69B'
   }
})
