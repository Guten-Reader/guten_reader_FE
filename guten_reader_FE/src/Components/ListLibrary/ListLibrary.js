import React from 'react';
import { View, Text, Button, StyleSheet } from 'react-native';
import { createAppContainer } from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';
import { withNavigation } from 'react-navigation';

function ListLibrary(props) {

  const handlePress = id => {
    props.downloadBook(1, id)
  }

  // needs to find book with this id and return currentText

  function displayBooks() {
    return props.books.map((book) => {
      return (
        <View key={book.id} style={styles.listCont}>
          <Text style={styles.line}>______________________________________________________</Text>
          <View style={styles.list}>
            <View>
              <Text style={styles.listItem, styles.title}>{book.title}</Text>
              <Text style={styles.listItem, styles.author}>{book.author}</Text>
            </View>
            <Button style={styles.button} onPress={() => handlePress(book.id)} title="READ"/>
          </View>
        </View>
      )
    })
  }

    return (
      <View>
        {displayBooks()}
      </View>
    )
}

const styles = StyleSheet.create({
  list: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 20
  },
  listItem: {
   fontSize: 20,
  },
   button: {
    fontSize: 20,
    color: '#53E69B'
  },
   listCont: {
     flexDirection: 'column',
  },
   line: {
     color: '#DDDDDD',
     paddingLeft: 20,
     marginBottom: -20,
   },
   title: {
     fontWeight: 'bold',
     fontSize: 20,
     paddingTop: 10,
     paddingBottom: 10,
     width: 200
   },
   author: {
     fontSize: 15
   }
})

export default withNavigation(ListLibrary);
