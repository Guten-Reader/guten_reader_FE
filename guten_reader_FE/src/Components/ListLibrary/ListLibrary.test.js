import React from 'react';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ListLibrary from './ListLibrary';

describe('ListLibrary renders correctly with Jest', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ListLibrary />);
    expect(wrapper).toMatchSnapshot();
  })
})