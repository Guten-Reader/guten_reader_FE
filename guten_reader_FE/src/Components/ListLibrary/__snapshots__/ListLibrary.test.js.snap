// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListLibrary renders correctly with Jest renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
